doc/use_cli_effectively.md

```diff
@@ -0,0 +1,88 @@
+# Tips of using Azure CLI effectively #
+
+## Use the right output mode for your work (json, table, or tsv) ##
+  1. Use `json` (CLI's default) to give you the most comprehensive information. If you prefer less, use `--output` argument to override at individual command level, or use "az configure" for global level.
+  2. Use `table` to give you less but much focused information. If some fields you care are missing, you have 2 options, use `--output json` to get all, or leverage `--query` to pick ones you like.
+  
+         az vm show -g my_rg -n my_vm --query "{name: name, os:storageProfile.imageReference.offer}" -otable
+         Name    Os
+         ------  ------------
+         my_vm   UbuntuServer
+         
+  3. Use `tsv` for the most concise output, which is good for scripting particularly you can save to variables to be used later, e.g.
+        
+        export vm_ids=$(az vm list -d -g my_rg --query "[?powerState=='VM running'].id" -o tsv)  
+        az vm stop --ids $vm_ids
+        
+## To pass values from one command to the other ##
+  1. If the value will be used more than once, assign it to a variable, e.g.
+       
+         running_vm_ids=$(az vm list -d -g my_rg --query "[?powerState=='VM running'].id" -o tsv)
```

Comment:
This is bash-specific syntax. We should call that out. What about people that use CMD.exe or Powershell?

Reply:
I added more examples at the end for cmd.exe and Powershell. I also called out at the beginning that examples inline are bash based.